external/contracts/DIAOracle.sol

```diff
@@ -0,0 +1,47 @@
+pragma solidity 0.6.10;
+import { IDIAOracle } from "../../contracts/interfaces/external/IDIAOracle.sol";
+
+contract DIAOracle is IDIAOracle{
+	address owner;
+
+	struct CoinInfo {
+		uint256 price;
+		uint256 supply;
+		uint256 lastUpdateTimestamp;
+		string symbol;
+	}
+
+	mapping(string => CoinInfo) diaOracles;
+	
+	event newCoinInfo(
+		string name,
+		string symbol,
+		uint256 price,
+		uint256 supply,
+		uint256 lastUpdateTimestamp
+	);
+    
+	constructor() public {
+		owner = msg.sender;
+	}
+
+	function changeOwner(address newOwner) public {
```

Comment:
Probably a matter of personal preference but could use Ownable from OpenZeppelin here to gut out the ownership logic.